app/controllers/carto/api/public/datasets_controller.rb

```diff
@@ -59,49 +51,56 @@ def enrich_tables(tables)
           table_names = tables.map { |table| table[:name] }
           visualizations = table_visualizations(table_names)
           tables.map do |table|
-            visualizations.find { |visualization| visualization[:name] == table[:name] } || table
+            table[:shared] = true unless table[:table_schema]==@user.username
+            viz = visualizations.find { |visualization| visualization[:name] == table[:name] } || { cartodbfied: false}
+            viz.merge(table)
           end
         end
 
         def table_visualizations(names)
           visualizations = Carto::VisualizationQueryBuilder.new
-                                                           .with_user_id(@user.id)
+                                                           .with_owned_by_or_shared_with_user_id(@user.id)
                                                            .with_name(names)
                                                            .with_type(Carto::Visualization::TYPE_CANONICAL)
                                                            .build.all
           visualizations.map do |visualization|
             {
               name: visualization.name,
-              cartodbfied: true,
-              type: 'table',
               privacy: visualization.privacy,
+              cartodbfied: true,
```

Comment:
Second part of the previous suggestion.
```suggestion
              cartodbfied: true,
              shared: !visualization.is_owner?(@user)
```